src/main.rs

```diff
@@ -2695,90 +2708,127 @@ async fn check_audit_as_crates_io(
             .map(|(name, version, _)| (name.clone(), version.cloned()))
             .collect();
 
-        for package in first_party_packages_strict(&cfg.metadata, &store.config) {
+        for package in &first_party_packages {
             // Remove both versioned and unversioned entries
             unused_audit_as.remove(&(package.name.clone(), Some(package.vet_version())));
             unused_audit_as.remove(&(package.name.clone(), None));
+        }
+        if !unused_audit_as.is_empty() {
+            errors.push(AuditAsError::UnusedAuditAs(UnusedAuditAsErrors {
+                errors: unused_audit_as
+                    .into_iter()
+                    .map(|(package, version)| PackageError { package, version })
+                    .collect(),
+            }))
+        }
+    }
 
-            let audit_policy = package
-                .policy_entry(&store.config.policy)
-                .and_then(|policy| policy.audit_as_crates_io);
-            if audit_policy == Some(false) {
-                // They've explicitly said this is first-party so we don't care about what's in the
-                // registry.
-                continue;
-            }
+    // We should only check the audit-as-crates-io entries if we have a network, because we
+    // shouldn't make recommendations based on potentially stale information.
+    if network.is_some() {
+        let progress = progress_bar(
+            "Fetching",
+            "crate metadata",
+            first_party_packages.len() as u64,
+        );
```

Comment:
A bit of me wonders if this should be more like `"Validating"`, `"audit-as-crates-io specifications"` so it's more clear why we're fetching metadata for these crates (especially because they're explicitly first-party crates). This name also works though.